src/main/java/reposense/model/CliArguments.java

```diff
@@ -1,34 +1,89 @@
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.util.List;
+import java.util.Objects;
 
 import reposense.parser.ArgsParser;
+import reposense.parser.AuthorConfigCsvParser;
+import reposense.parser.GroupConfigCsvParser;
+import reposense.parser.RepoConfigCsvParser;
+import reposense.parser.ReportConfigJsonParser;
 
 /**
  * Represents command line arguments user supplied when running the program.
  */
-public abstract class CliArguments {
-    protected Path outputFilePath;
-    protected Path assetsFilePath;
-    protected LocalDateTime sinceDate;
-    protected LocalDateTime untilDate;
-    protected boolean isSinceDateProvided;
-    protected boolean isUntilDateProvided;
-    protected List<FileType> formats;
-    protected boolean isLastModifiedDateIncluded;
-    protected boolean isShallowCloningPerformed;
-    protected boolean isAutomaticallyLaunching;
-    protected boolean isStandaloneConfigIgnored;
-    protected boolean isFileSizeLimitIgnored;
-    protected int numCloningThreads;
-    protected int numAnalysisThreads;
-    protected ZoneId zoneId;
-    protected boolean isFindingPreviousAuthorsPerformed;
-    protected boolean isTestMode = ArgsParser.DEFAULT_IS_TEST_MODE;
-    protected boolean isFreshClonePerformed = ArgsParser.DEFAULT_SHOULD_FRESH_CLONE;
+public class CliArguments {
+    private static final Path EMPTY_PATH = Paths.get("");
+    private static final Path DEFAULT_CONFIG_PATH = Paths.get(System.getProperty("user.dir")
+            + File.separator + "config" + File.separator);
+
+    private final Path outputFilePath;
+    private final Path assetsFilePath;
+    private final LocalDateTime sinceDate;
+    private final LocalDateTime untilDate;
+    private final boolean isSinceDateProvided;
+    private final boolean isUntilDateProvided;
+    private final List<FileType> formats;
+    private final boolean isLastModifiedDateIncluded;
+    private final boolean isShallowCloningPerformed;
+    private final boolean isAutomaticallyLaunching;
+    private final boolean isStandaloneConfigIgnored;
+    private final boolean isFileSizeLimitIgnored;
+    private final int numCloningThreads;
+    private final int numAnalysisThreads;
+    private final ZoneId zoneId;
+    private final boolean isFindingPreviousAuthorsPerformed;
+    private boolean isTestMode = ArgsParser.DEFAULT_IS_TEST_MODE;
+    private boolean isFreshClonePerformed = ArgsParser.DEFAULT_SHOULD_FRESH_CLONE;
+
+
+    // ViewCliArguments.java
+    private List<String> locations;
+
+    // LocationsCliArguments.java
+    private Path reportDirectoryPath;
+
+    // ConfigCliArguments.java
+    private Path configFolderPath;
+    private Path repoConfigFilePath;
+    private Path authorConfigFilePath;
+    private Path groupConfigFilePath;
+    private Path reportConfigFilePath;
+    private ReportConfiguration reportConfiguration;
+
+    public CliArguments(Builder builder) {
```

Comment:
Since the builder is used to create the CliArguments (and I believe that should be the only way), do you think it is better for this constructor be `protected` or have its access limited?

Reply:
Good catch, I'll update it to private.